sdk/core/core-http/lib/util/xml.browser.ts

```diff
@@ -1,16 +1,32 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+import { removeBOM } from "./utils";
+
 // tslint:disable-next-line:no-null-keyword
 const doc = document.implementation.createDocument(null, null, null);
 
 const parser = new DOMParser();
-export function parseXML(str: string): Promise<any> {
+export function parseXML(str: string, opts?: { includeRoot?: boolean }): Promise<any> {
   try {
+    if (str == undefined || str == "") {
+      throw new Error("Document is empty");
```

Comment:
This is a new error case -- what used to happen when you passed and empty XML string before? 

I would prefer that we match the existing behavior so that we don't break existing scenarios.

Reply:
This is actually the same error that gets thrown by parser in Node - same for all errors in .browser.ts file

Reply:
The error message is not what I was wondering. If another SDK was using this browser code before and they passed either `undefined` or and empty document, what would they get back? 

I am asking because changing that behavior (and throwing in that case) could be a breaking change.

Reply:
Based on usage in Storage SDK, this scenario doesn't seem to exist
It seems a very less likely possibility since the SDKs are all targeting the same Azure services. And for an agnostic thing such as empty/undefined document being disallowed in one SDK, it seems like the behavior should be same for others as well.

For instance, if parsing of undefined/empty documents is allowed in browser, it would be a bug if it doesn't work from the Node SDK?
Similarly, if it is not allowed in browser, then the tests and checks would have helped catch that type of bug early on.

But I think more investigation is needed on this, since maybe we should allow handling of empty/undefined documents if that is a valid scenario? Ideally, this discrepancy wouldn't be there if we are using the same parser for Node and browser, but for most part the behaviors would be expected to be same unless very explicitly required as otherwise I think..
cc @bterlson @daviwil